backend/de.metas.elasticsearch/src/main/java/de/metas/fulltextsearch/indexer/handler/FTSModelIndexerRegistry.java

```diff
@@ -23,51 +23,50 @@
 package de.metas.fulltextsearch.indexer.handler;
 
 import com.google.common.collect.ImmutableList;
-import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableListMultimap;
 import de.metas.logging.LogManager;
-import de.metas.util.GuavaCollectors;
 import lombok.NonNull;
 import org.adempiere.ad.table.api.TableName;
 import org.slf4j.Logger;
 import org.springframework.stereotype.Component;
 
 import java.util.Collection;
+import java.util.HashSet;
 import java.util.List;
-import java.util.Objects;
 import java.util.Optional;
+import java.util.Set;
 
 @Component
 public class FTSModelIndexerRegistry
 {
 	private static final Logger logger = LogManager.getLogger(FTSModelIndexerRegistry.class);
 
-	private final ImmutableMap<TableName, FTSModelIndexer> indexersBySourceTableName;
+	private final ImmutableListMultimap.Builder<TableName, FTSModelIndexer> indexersBySourceTableName = ImmutableListMultimap.builder();
 
 	public FTSModelIndexerRegistry(@NonNull final Optional<List<FTSModelIndexer>> indexers)
 	{
-		this.indexersBySourceTableName = indexers.orElseGet(ImmutableList::of)
-				.stream()
-				.flatMap(indexer -> indexer.getHandledSourceTableNames()
-						.stream()
-						.map(sourceTableName -> GuavaCollectors.entry(sourceTableName, indexer)))
-				.collect(GuavaCollectors.toImmutableMap());
+		for (final FTSModelIndexer indexer : indexers.orElseGet(ImmutableList::of))
```

Comment:
this.indexersBySourceTableName = indexers.orElseGet(ImmutableList::of)
.stream()
.collect(ImmutableListMultimap.toImmutableListMultimap(.....))